vowpalwabbit/allreduce_sockets.cc

```diff
@@ -16,10 +16,11 @@ Alekh Agarwal and John Langford, with help Olivier Chapelle.
 #include <stdlib.h>
 #ifdef _WIN32
 #define NOMINMAX
-#include <WinSock2.h>
-#include <Windows.h>
-#include <WS2tcpip.h>
-#include <io.h>
+#  define _WINSOCK_DEPRECATED_NO_WARNINGS
```

Comment:
`_WINSOCK_DEPRECATED_NO_WARNINGS` removed from `vcxproj` files and added here so that when building with `cmake` the warning is also silenced